api/src/test/java/org/openmrs/api/db/hibernate/HibernateConditionDAOTest.java

```diff
@@ -0,0 +1,138 @@
+import static org.junit.Assert.assertNull;
+
+import java.util.Date;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.openmrs.*;
+import org.openmrs.api.db.ConditionDAO;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class HibernateConditionDAOTest extends BaseContextSensitiveTest {
+	
+	private static final String CONDITIONS_XML = "org/openmrs/api/db/hibernate/include/HibernateConditionDAOTestDataSet.xml";
+	
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+	
+	@Autowired
+	ConditionDAO dao;
+	
+	
+	@Before
+	public void setUp() {
+		executeDataSet(CONDITIONS_XML);
+		
+		updateSearchIndex();
+	}
+	
+	@Test
+	public void shouldSaveCondition() {
+		CodedOrFreeText codedOrFreeText = new CodedOrFreeText(new Concept(4),
+			new ConceptName(5089), "non coded");
+		ConditionClinicalStatus clinicalStatus = ConditionClinicalStatus.ACTIVE;
+		ConditionVerificationStatus verificationStatus = ConditionVerificationStatus.CONFIRMED;
+		Patient patient = new Patient(2);
+		Date onsetDate = new Date();
+		Date endDate = new Date();
+		Condition previousVersion = dao.getConditionByUuid("2cc6880e-2c46-15e4-9038-a6c5e4d22fb7");
+		String additionalDetail = "additionalDetail";
+		
+		Condition condition = new Condition();
+
+		condition.setConditionId(20);
+		condition.setCondition(codedOrFreeText);
+		condition.setClinicalStatus(clinicalStatus);
+		condition.setVerificationStatus(verificationStatus);
+		condition.setPreviousVersion(previousVersion);
+		condition.setAdditionalDetail(additionalDetail);
+		condition.setOnsetDate(onsetDate);
+		condition.setEndDate(endDate);
+		condition.setPatient(patient);
+		
+		Condition savedCondition = dao.saveCondition(condition);
+		
+		assertEquals(additionalDetail, savedCondition.getAdditionalDetail());
+		assertEquals(20, (int) savedCondition.getConditionId());
+		assertEquals(onsetDate, (Date) savedCondition.getOnsetDate());
+		assertEquals(endDate, savedCondition.getEndDate());
+		assertEquals(clinicalStatus, savedCondition.getClinicalStatus());
+		assertEquals(verificationStatus, savedCondition.getVerificationStatus());
+		assertEquals(previousVersion, (Condition) savedCondition.getPreviousVersion());
+		assertEquals(patient, (Patient) savedCondition.getPatient());
+	}
+	
+	@Test
+	public void shouldGetConditionByUuid() {
+		String uuid = "2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		ConditionClinicalStatus expectedClinicalStatus = ConditionClinicalStatus.INACTIVE;
+		ConditionVerificationStatus expectedVerificationStatus = ConditionVerificationStatus.PROVISIONAL;
+		
+		Condition condition = dao.getConditionByUuid(uuid);
+		assertEquals(condition.getClinicalStatus(), expectedClinicalStatus);
+		assertEquals(condition.getVerificationStatus(), expectedVerificationStatus);
+		assertEquals(1, (int) condition.getId());
```

Comment:
Is all this casting necessary?

Reply:
The compiler complains of ambiguous types if casting is not done